### PR TITLE
Fix flaky `train_small` being 2X slow 

### LINKED ISSUE #19305: [Nightly] train_small timeout. Stuck in `Waiting until actors are ready` for more than 60seconds
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Others

### What happened + What you expected to happen

train_small is timing out. Most notably `waiting untill actors are ready` takes more than 60s. This is after `wait_cluster` is done so no more autoscaling should be happening.
Happens both in the case using ray client and not.

```
[INFO 2021-10-10 11:23:56,465] e2e.py: 825  Starting job train_small_1633889125 with Ray address: anyscale://train_small_1633889125?autosuspend=10
--
  | (pid=1279) 2021-10-10 11:24:23,274	INFO main.py:913 -- [RayXGBoost] Created 4 new actors (4 total actors). Waiting until actors are ready for training.
  | The actor or task with ID ffffffffffffffff625c288d32dc2a41fdedf2a402000000 cannot be scheduled right now. You can ignore this message if this Ray cluster is expected to auto-scale or if you specified a runtime_env for this actor or task, which may take time to install.  Otherwise, this is likely due to all cluster resources being claimed by actors. To resolve the issue, consider creating fewer actors or increasing the resources available to this Ray cluster.
  | Required resources for this actor or task: {CPU_group_cda3ad2861590c44996ec6a82e0b2cfc: 4.000000}
  | Available resources on this node: {0.000000/4.000000 CPU, 587202559.960938 GiB/587202559.960938 GiB memory, 233156159.960938 GiB/233156159.960938 GiB object_store_memory, 0.000000/4.000000 CPU_group_cda3ad2861590c44996ec6a82e0b2cfc, 1.000000/1.000000 node:172.31.120.249, 1000.000000/1000.000000 bundle_group_2_cda3ad2861590c44996ec6a82e0b2cfc, 1000.000000/1000.000000 bundle_group_cda3ad2861590c44996ec6a82e0b2cfc, 4.000000/4.000000 CPU_group_2_cda3ad2861590c44996ec6a82e0b2cfc}
  | In total there are 0 pending tasks and 1 pending actors on this node.
  | (pid=1279) 2021-10-10 11:24:53,302	INFO main.py:947 -- Waiting until actors are ready (30 seconds passed).
  | (pid=1279) 2021-10-10 11:25:23,375	INFO main.py:947 -- Waiting until actors are ready (60 seconds passed).
  | (pid=1279) 2021-10-10 11:25:36,855	INFO main.py:958 -- [RayXGBoost] Starting XGBoost training.
```

Link: https://buildkite.com/ray-project/periodic-ci/builds/1209#49028e00-affd-404f-b2b3-7309a875787b

### Reproduction script

Nightly release test

### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

TL;DR: This was due to the Ray core issue, but it won't be fixed in this PR. This PR makes tests pass consistently by "avoiding" scenarios where Ray core has a bug. Core bug will be fixed by me in next 2 weeks.

The issue was that the cluster has 16 CPUs, and the placement group requires 16 CPUs. But we are also requesting 1 CPU for invoking a remote function `train`. 

This happened to work due to Ray's "CPU borrowance" feature. Basically, when `ray.get(pg.ready())` was called, it made 1 CPU available which makes pg able to be scheduled. Look below for more details.

### Step 1
Original cluster
{CPU: 4}
{CPU: 4}
{CPU: 4}
{CPU: 4}

### Step 2
`train` remote function scheduled
{CPU: 4}
{CPU: 4}
{CPU: 4}
{CPU: 3} # 1 CPU used by train

### Step 3
PG is created and pending since it requires 16 CPUs.

### Step 4
`ray.get(pg.ready()` is called (from the Xgboost on Ray I believe), and 1 CPU is temporarily available (Ray releases CPU temporarily when blocking API is called to avoid deadlock). So, pg is scheduled since 16 CPUs are available.

{CPU: 0} # 4 CPUs Used by pg
{CPU: 0} # 4 CPUs Used by pg
{CPU: 0} # 4 CPUs Used by pg
{CPU: -1} # 4 CPUs Used by pg + 1 CPU used by `train`

Here, the Ray core has a bug where in this scenario, the resource is incorrectly propagated, which made tests slow. The PR **didn't fix that issue yet**, but it will make `train_small` at least pass consistently by removing CPU request for `train`. I will fix the issue where the resource propagation is done incorrectly within next 2 weeks. 

## Related issue number

Closes https://github.com/ray-project/ray/issues/19305. I will open a separate issue

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by krfricke — APPROVED** [review]
Thanks! 

I created a release test run here: https://buildkite.com/ray-project/periodic-ci/builds/1706

I'll merge after this has passed.
